internal/controller/controllers/agent_controller.go

```diff
@@ -317,6 +317,16 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 				return ctrl.Result{Requeue: true}, nil
 			}
 		}
+		if agent.Status.DebugInfo.LogsURL == "" && !time.Time(h.LogsCollectedAt).Equal(time.Time{}) { // logs collection time is updated means logs are available
+			logsURL, err := generateControllerLogsDownloadURL(r.ServiceBaseURL, clusterId.String(), r.AuthType, agent.Name, "host")
+			if err != nil {
+				log.WithError(err).Error("failed to generate controller logs URL")
+				return ctrl.Result{}, err
+			}
+			if logsURL != "" {
```

Comment:
Just noticed this. Will this ever return and empty string and no error?
Feels like this shouldn't be necessary.

Reply:
True! I've removed it. Thank you!